source/vibe/core/core.d

```diff
@@ -877,18 +892,64 @@ unittest {
 
 
 /**
-	Returns a new armed timer.
-
-	Note that timers can only work if an event loop is running, explicitly or
-	implicitly by running a blocking operation, such as `sleep` or `File.read`.
+	Creates a new timer, that will fire `callback` after `timeout`
+
+    There are two kinds of timer: one-off timers and periodic timers.
+    One-off timers fire only once, after a specific amount of time,
+    while periodic timer fire at a regular interval.
+
+    One-off_timers:
+    One-off timers can be used for performing a task after a specific delay,
+    or for the initial setup of an application. For example, the following
+    is a common pattern for a Vibe.d application:
```

Comment:
Hum, right. Actually one primitive I am missing from `std.concurrency` is `schedule`, that is, a `runTask` that doesn't switch immediately, which we currently use `Timer`  with a 0 timeout for.

An example use case: We get data, that we need to propagate. Once we validated it, we schedule a network request to our peer, then do other operations (updating local state, storing on disk, etc...).
When we used `runTask` originally, what would happen was that the nodes would DOS themselves, because they would immediately propagate it, then other nodes would receive it and immediately propagate it, and we might get the new propagation back before we're even done processing the data we originally received (once the processing is done, we ignore duplicated data).

I'll update the example with the `schedule` approach in mind.

Reply:
True, I had the need for that in the past, too, but always resorted to `yield()` at the start of the task, because it seemed to be a niche case. But I introduced `TaskSettings` a while ago, that would lend itself very well for a `deferred` flag.